src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -89,43 +106,56 @@ BackupCoordinationStageSync::BackupCoordinationStageSync(
     , max_attempts_after_bad_version(with_retries.getKeeperSettings().max_attempts_after_bad_version)
     , zookeeper_path(zookeeper_path_)
     , root_zookeeper_path(zookeeper_path.parent_path().parent_path())
-    , operation_node_path(zookeeper_path.parent_path())
+    , operation_zookeeper_path(zookeeper_path.parent_path())
     , operation_node_name(zookeeper_path.parent_path().filename())
-    , stage_node_path(zookeeper_path)
     , start_node_path(zookeeper_path / ("started|" + current_host))
     , finish_node_path(zookeeper_path / ("finished|" + current_host))
     , num_hosts_node_path(zookeeper_path / "num_hosts")
+    , error_node_path(zookeeper_path / "error")
     , alive_node_path(zookeeper_path / ("alive|" + current_host))
     , alive_tracker_node_path(fs::path{root_zookeeper_path} / "alive_tracker")
-    , error_node_path(zookeeper_path / "error")
     , zk_nodes_changed(std::make_shared<Poco::Event>())
 {
-    if ((zookeeper_path.filename() != "stage") || !operation_node_name.starts_with(is_restore ? "restore-" : "backup-")
-        || (root_zookeeper_path == operation_node_path))
-    {
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected path in ZooKeeper specified: {}", zookeeper_path);
-    }
-
     initializeState();
     createRootNodes();
 
     try
     {
+        concurrency_check.emplace(is_restore, /* on_cluster = */ true, zookeeper_path, allow_concurrency, concurrency_counters_);
```

Comment:
If the concurrency check we have to `setError()` to tell other hosts about it.  That wasn't done before and that's why tests `test_disallow_concurrency` were flaky.